serverless/email/webhook/main.go

```diff
@@ -48,17 +52,52 @@ var (
 	sesClient            *ses.Client
 	secretsManagerClient *secretsmanager.Client
 
-	namespaceSecretOutput *secretsmanager.GetSecretValueOutput
-	authTokenSecretOutput *secretsmanager.GetSecretValueOutput
+	namespaceSecretOutput   *secretsmanager.GetSecretValueOutput
+	authTokenSecretOutput   *secretsmanager.GetSecretValueOutput
+	authSecretsSecretOutput *secretsmanager.GetSecretValueOutput
+
+	// header names
+	tsHeaderName  = os.Getenv("TS_HEADER_NAME")
+	sigHeaderName = os.Getenv("SIG_HEADER_NAME")
+	keyHeaderName = os.Getenv("KEY_HEADER_NAME")
 )
 
+// validateSignature - validate the signature provided from partner
+func validateSignature(request events.APIGatewayProxyRequest) error {
+	// signingString => {ts}{method}{path}{body}
+	signingString := fmt.Sprintf("%s%s%s%s",
+		request.Headers[tsHeaderName], request.HTTPMethod, request.Path, request.Body)
+	signatureBytes, err := hex.DecodeString(request.Headers[sigHeaderName])
+	if err != nil {
+		return fmt.Errorf("failed to decode signature from headers: %w", err)
+	}
+
+	var valid bool
+	// sha256 hmac(apiSecret, signingString)
+	for _, apiSecret := range strings.Split(*authTokenSecretOutput.SecretString, ",") {
```

Comment:
Crypto implementation wise this looks fine, but I'm a bit confused about the usage for the for loop here. Is it not known which API Key is going to be in use for the key here?

Reply:
In order to support multiple signing keys for partner key rotation we decided to perform this check in a loop over the range of the signing keys we maintain with the partner.

Reply:
Ok seems like a reasonable tradeoff here. My only concern was that it would slow down verification time for every request, but if that actually is the case it's an easy refactor later. Resolving this one